tests/gatewayapitests/status/httproute_test.go

```diff
@@ -1019,3 +1030,77 @@ func TestHttpRouteWithValidAndInvalidGatewayListeners(t *testing.T) {
 	akogatewayapitests.TeardownGateway(t, gatewayName, namespace)
 	akogatewayapitests.TeardownGatewayClass(t, gatewayClassName)
 }
+func TestHTTPRouteWithInvalidBackendKind(t *testing.T) {
+	gatewayClassName := "gateway-class-hr-16"
+	gatewayName := "gateway-hr-16"
+	httpRouteName := "httproute-16"
+	namespace := "default"
+	ports := []int32{8080}
+
+	akogatewayapitests.SetupGatewayClass(t, gatewayClassName, akogatewayapilib.GatewayController)
+
+	listeners := akogatewayapitests.GetListenersV1(ports, false, false)
+	akogatewayapitests.SetupGateway(t, gatewayName, namespace, gatewayClassName, nil, listeners)
+
+	g := gomega.NewGomegaWithT(t)
+	g.Eventually(func() bool {
+		gateway, err := akogatewayapitests.GatewayClient.GatewayV1().Gateways(namespace).Get(context.TODO(), gatewayName, metav1.GetOptions{})
+		if err != nil || gateway == nil {
+			t.Logf("Couldn't get the gateway, err: %+v", err)
+			return false
+		}
+		return apimeta.FindStatusCondition(gateway.Status.Conditions, string(gatewayv1.GatewayConditionAccepted)) != nil
+	}, 30*time.Second).Should(gomega.Equal(true))
+
+	parentRefs := akogatewayapitests.GetParentReferencesV1([]string{gatewayName}, namespace, ports)
+	rule := akogatewayapitests.GetHTTPRouteRuleV1([]string{"/foo"}, []string{},
+		map[string][]string{"RequestHeaderModifier": {"add", "remove", "replace"}},
+		[][]string{{"avisvc", "default", "8080", "1"}}, nil)
+	kind := gatewayv1.Kind("InvalidKind")
+	rule.BackendRefs[0].BackendRef.Kind = &kind
```

Comment:
Can we have additional unit test with two backends, one invalid and one valid and then check for Conditions

Reply:
Done